motioneye/uploadservices.py

```diff
@@ -845,6 +846,87 @@ def _request_credentials(self, authorization_key):
         return {'access_token': data['access_token']}
 
 
+class Nextcloud(UploadService):
+    NAME = 'nextcloud'
+
+    def __init__(self, camera_id):
+        self._server = None
+        self._port = None
+        self._username = None
+        self._password = None
+        self._location = None
+
+        UploadService.__init__(self, camera_id)
+
+    def _get_base_url(self):
+        scheme = 'http://'
+        if self._port == 443:
+            scheme = 'https://'
+        url = scheme + self._server + ':' + str(self._port) + '/remote.php/dav/files/' + self._username + '/'
+        return url
+
+    def _request(self, url, method, data=None):
+        self.debug('request: ' + method + ' ' + url)
+        request = urllib2.Request(url, data=data)
+        request.get_method = lambda: method
+        base64string = base64.b64encode('%s:%s' % (self._username, self._password))
+        request.add_header("Authorization", "Basic %s" % base64string)
+        if data is not None:
+            request.add_header('Content-Length', '%d' % len(data))
+        try:
+            utils.urlopen(request)
+        except urllib2.HTTPError as e:
+            if method == 'MKCOL' and e.code == 405:
+                self.debug('MKCOL failed with code 405, this is normal if the folder exists')
+            else:
+                raise e
+
+    def _make_dirs(self, path):
+        dir_url = self._get_base_url()
+        for folder in path.strip('/').split('/'):
+            dir_url = dir_url + folder + '/'
+            self._request(dir_url, 'MKCOL')
+
+    def test_access(self):
+        try:
+            test_path = self._location.strip('/') + '/' + str(time.time())
+            self._make_dirs(test_path)
+            self._request(self._get_base_url() + test_path, 'DELETE')
+            return True
+        except Exception as e:
+            self.error(str(e), exc_info=True)
+            return str(e)
+
+    def upload_data(self, filename, mime_type, data, ctime, camera_name):
+        path = self._location.strip('/') + '/' + os.path.dirname(filename) + '/'
+        filename = os.path.basename(filename)
+        self._make_dirs(path)
+        self.debug('uploading %s of %s bytes' % (filename, len(data)))
+        self._request(self._get_base_url() + path + filename, 'PUT', bytearray(data))
```

Comment:
f-strings are faster and easier to read.
```suggestion
        path = f"{self._location.strip('/')}/{os.path.dirname(filename)}/"
        filename = os.path.basename(filename)
        self._make_dirs(path)
        self.debug(f"uploading {filename} of {len(data)} bytes")
        self._request(f"{self._get_base_url()}{path}{filename}", 'PUT', bytearray(data))
```

Reply:
Suggesting single quotes `'` instead of double quotes, for consistency. Or are double quotes required or any specific kind of expansion here?